whois-api/src/main/java/net/ripe/db/whois/api/httpserver/RewriteEngine.java

```diff
@@ -27,18 +27,27 @@ public class RewriteEngine {
     private final String source;
     private final String nonAuthSource;
 
+    private final String clientAuthHost;
+
     @Autowired
     public RewriteEngine(final @Value("${api.rest.baseurl}") String baseUrl,
                          final @Value("${whois.source}") String source,
-                         final @Value("${whois.nonauth.source}") String nonAuthSource) {
+                         final @Value("${whois.nonauth.source}") String nonAuthSource,
+                         final @Value("${api.client.auth.baseurl}") String clientAuthBaseUrl) {
```

Comment:
Make this String optional, i.e. it doesn't have to be set if `client.auth.enabled` is false. Also check for an empty String below.